kbase-extension/static/kbase/js/common/jobs.js

```diff
@@ -104,6 +105,92 @@ define(['common/errorDisplay', 'common/format', 'common/html', 'common/ui', 'uti
         return jobStrings[status][stringType];
     }
 
+    /**
+     *
+     * @param {object} args with keys
+     *  message - the message for validation
+     *  type    - the message type (one of jcm.MESSAGE_TYPEs)
+     *
+     * @returns {object} with keys 'valid' and 'invalid' containing valid/invalid data
+     *                  if the whole message is invalid, there will be no 'valid' key,
+     *                  and vice versa for a valid message
+     */
+    function validateMessage(args) {
+        const { message, type } = args;
+
+        const validationFn = {
+            [jcm.RESPONSES.INFO]: isValidJobInfoObject,
+            [jcm.RESPONSES.LOGS]: isValidJobLogsObject,
+            [jcm.RESPONSES.STATUS]: isValidBackendJobStateObject,
+            [jcm.RESPONSES.STATUS_ALL]: isValidBackendJobStateObject,
+            [jcm.RESPONSES.RETRY]: isValidJobRetryObject,
+            [jcm.RESPONSES.RUN_STATUS]: isValidRunStatusObject,
```

Comment:
No validator for `NEW` messages, but since they just trigger a narrative save, there is not much to worry about. Also missing validators for the `CELL_JOB_STATUS` and `ERROR` messages; the former are not yet in use but are in the same format as `STATUS` messages, so that validator can be reused; the latter really need to be collated and recorded to ensure that the FE is responding to the appropriate data structures.